libraries/apollo-runtime/src/jsMain/kotlin/com/apollographql/apollo3/network/http/KtorHttpEngine.kt

```diff
@@ -17,10 +17,16 @@ import io.ktor.util.flattenEntries
 import io.ktor.utils.io.CancellationException
 import okio.Buffer
 
-
+/**
+ * @param connectTimeoutMillis The connection timeout in milliseconds. The connection timeout is the time period in which a client should establish a connection with a server.
+ * @param readTimeoutMillis The request timeout in milliseconds. The request timeout is the time period required to process an HTTP call: from sending a request to receiving a response.
+ */
 actual class DefaultHttpEngine constructor(private val connectTimeoutMillis: Long, private val readTimeoutMillis: Long) : HttpEngine {
```

Comment:
While we're at it, do you mind renaming `readTimeoutMillis` to `requestTimeoutMillis`? Ultimately, I'd like us to support `readTimeoutMillis` because:

* I think it's a better parameter because it does not depend on the response size
* `requestTimeout` can be emulated quite easily using `withTimeout` while `readTimeoutMillis` actually requires low-level configuration

Reply:
So all in all it'd be:

* `connectTimeout`: the initial TCP handshake timeout
* `readTimeout`: the socket timeout on subsequent reads
* `requestTimeout` (not very useful): the overall timeout of the request

Reply:
I think renaming `readTimeoutMillis` to `requestTimeoutMillis` is breaking right? 

Also, in my tests, `readTimeoutMillis` didn't seem to work actually 😅 But I'll try again to confirm.

In any case I think it makes sense to add a constructor with the 3 parameters? Give more power to the user.

Reply:
> I think renaming readTimeoutMillis to requestTimeoutMillis is breaking right?

Indeed, source breaking, which unfortunately isn't tracked by [kbcv](https://github.com/Kotlin/binary-compatibility-validator). 

> In any case I think it makes sense to add a constructor with the 3 parameters? Give more power to the user.

I'd vote for only 2: `connectTimeout` and `readTimeout`. `requestTimeout` belongs in higher layers using [withTimeout](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-timeout.html). That is if `readTimeout` is working as expected of course.

Reply:
I just checked and `socketTimeout` doesn't work on Ktor JS. I'll open an issue. In the meantime we could use `socketTimeout` for Jvm and Apple, but `requestTimeout` for JS?

Reply:
I think so? If you're talking about the expect/actual constructor that takes a simple `timeoutMillis` parameter then it's fine to map it to `socketTimeout` on Jvm/Apple and `requestTimeout` on JS as long as it's properly documented.

Then for overloaded (non-actual) constructors, my preference would go to `connectTimeout` and `readTImeout` and leave `requestTimeout` out of it.